File: backend/src/main/java/com/pvreview/model/Case.java
package com.pvreview.model;

import java.util.List;
import java.util.Map;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Case {

    private String caseId;
    private int version;
    private String caseClassification;
    private String extractedAt;
    private String sourceDocument;
    private Map<String, Map<String, FieldRecord>> sections;
    private List<String> missingFields;
}

File: frontend/src/types/case.ts
export type FieldStatus = 'UNCHANGED' | 'OVERRIDDEN' | 'NEW' | 'RETAINED'

export interface CaseField<T = string> {
  value: T
  confidence: number
  source: string
  status: FieldStatus
  /** Present when status is OVERRIDDEN. */
  previous_value?: T
}

export type CaseSection = Record<string, CaseField>

export interface CaseSections {
  patient: CaseSection
  suspect_drug: CaseSection
  adverse_event: CaseSection
  reporter: CaseSection
  [section: string]: CaseSection
}

export interface PvCase {
  case_id: string
  version: number
  case_classification: string
  extracted_at: string
  source_document: string
  sections: CaseSections
  missing_fields: string[]
}
